docker/lib/dependabot/docker/file_parser.rb

```diff
@@ -154,6 +160,68 @@ def check_required_files
 
         raise "No Dockerfile!"
       end
+
+      def workfile_file_dependencies(file)
```

Comment:
A k8s yaml template file may contain [multiple resource definitions](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#organizing-resource-configurations) separated by `---`, so we should make sure we can correctly handle that. By my testing, when given such a file Psych will parse the first one and ignore all the rest. That's acceptable for the purposes of the heuristic above, but here I think we'll want to make sure we handle all of them with something like

```ruby
def workfile_file_dependencies(file)
  dependency_set = DependencySet.new

  resources = file.content.split(/^---$/).map(&:strip).reject(&:empty?) # assuming a yaml file
  resources.flat_map do |resource|
    json = YAML.safe_load(resource, aliases: true)
    # ...
```

Reply:
done.